### ISSUE TITLE
Android mobile app not working with Piwigo 13

### ISSUE THREAD

**Issue by axb21** [Piwigo/piwigo-flutter-app#134]
Ever since Piwigo 11->13, both Piwigo Android apps have stopped working for me.

I can see a list of my albums. However, photos are not displayed, only blank squares where the photos should be. 

I cannot upload photos. I'm given an error that the upload failed. 

I've googled around and haven't found any helpful solutions. So I'm posting here.

**Comment by remi-martin** [comment]
Hi, please can you tell me your version of Piwigo NG ?
Also, are you using Community plugin ?
And, is it a piwigo.com server or a self hosted Piwigo ?

Those information will help a lot.

**Comment by axb21** [comment]
Hi, thanks for responding!

- Piwigo NG 1.2.1
- Piwigo server 13.1.0, self hosted
- I do not have the community plugin installed (though everything worked fine in the past without it)

Happy to share any other info that might help.

**Comment by remi-martin** [comment]
Hi, as far as I know, this issue also happens on Piwigo Mobile (iOS app with a different language). I also talked with the Piwigo team but they don't know where this issue comes from.

So I might need an access to your gallery so I can see the app logs about this issue. Please can you give me the url of your Piwigo with a test account ? (no specific rights needed as I just need to see the gallery)

**Comment by remi-martin** [comment]
Please can you tell me what is the php version you are using

**Comment by axb21** [comment]
```bash
PHP 8.1.12 (cli) (built: Oct 28 2022 17:39:18) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.12, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.12, Copyright (c), by Zend Technologies
```

**Comment by remi-martin** [comment]
Piwigo 13 added support to PHP 8, which might have generated bugs.
In order to have more info about this issue, can you give me the url and an access to your Piwigo Server ?

**Comment by axb21** [comment]
> Piwigo 13 added support to PHP 8, which might have generated bugs. In order to have more info about this issue, can you give me the url and an access to your Piwigo Server ?

No, I am not comfortable using that. It's a personal server for myself and my family. I don't want to publicize the URL and I don't want to create accounts for people I don't know. Please let's come up with a better way to proceed.

**Comment by remi-martin** [comment]
Unfortunately we don't.
Can you create another Piwigo on your server with the same configuration (empty) ?

**Comment by axb21** [comment]
> Unfortunately we don't. Can you create another Piwigo on your server with the same configuration (empty) ?

No, I don't want to share the URL of my server with a stranger on Github. 

Obviously there are other ways to debug problems than having access to someone's server. Am I the only person having this issue? Are there logs I can look in or send you? Does the app have a debug mode I can enable? There must be something.

**Comment by axb21** [comment]
I am still having this issue. If you could supply me with an email address or some other off-band way to contact you, I ca send you a URL/gallery to tinker with. I'd really like to get this fixed because at the moment Piwigo is non-functional from phones, and web uploads are cumbersome. I've half a mind to switch servers because of this, which feels so wrong.

**Comment by remi-martin** [comment]
Hi, sorry for the delay.
As far as we know, there is an issue with migration to Piwigo 13 (see #135).
Concerning the upload on the web version, I'll recommend to contact the web team on [Github](https://github.com/Piwigo/Piwigo) on via the [forums](https://piwigo.org/forum/).

**Comment by axb21** [comment]
OK, I see. I double checked my database and I do have the `user_agent` column mentioned in #135 so I don't think I'm having that particular issue. I'll check the forums. 

It'd be nice to know if the issue is in the server or in the Android app (or both), though. Any thoughts on that?

**Comment by remi-martin** [comment]
I can't be sure, but since the issue appeared after the migration and concerns image display and upload, I think the itis server sided. This is the major issue with developping a tier application for an infinity self-hosted websites: you can't handle every configuration, there will always be a problem and sometimes the app can't fix it.

**Comment by axb21** [comment]
OK, I have sorted out this problem and you can close this issue whenever you feel like it.

The problem turned out to be a known issue with running the Piwigo server behind a reverse proxy, which I do. I have no idea why I didn't have this problem in the past, which is why I was confused and thought the problem might have been the client. However, it was definitely that. I made a change suggested in this issue: https://github.com/Piwigo/Piwigo/issues/982#issuecomment-986004313 and everything works as expected now.

If it helps anyone else, the way I figured this out was the issue is I figured out the HTTP request being made to the server by the client by looking in my web server logs, and then ran that query in Chrome so I could see what was being sent back. I could see that Piwigo was adding the port number to the URLs when it should not. I could also verify that once I applied the above change to my configuration the port number was removed.

**Comment by remi-martin** [comment]
I'll add logs showing the url on images when they cannot be show so this kind of issue can be resolved faster.

**Comment by hardwareadictos** [comment]
Hey! I fixed it long ago with the mentioned fix: https://github.com/Piwigo/Piwigo/issues/982#issuecomment-986004313 but for some reason it is not working again behing a reverse proxy. Anyone on the same situation?
